src/hooks/usePreferences.js

```diff
@@ -0,0 +1,38 @@
+import { useCallback, useRef } from 'react'
+import { useHistory, useLocation } from 'react-router-dom'
+import { getPreferencesSearch } from '../Routes'
+
+/**
+ * Hook to interact with the preferences
+ * @returns {Array} first parameter open preferences handler, second parameter close preferences handler
+ *  third parameter the selected preference option
+ */
+export default function usePreferences() {
+  /* We need to keep track of the path where the preference was called 
+    in order to return to the same path when the preference modal is closed
+  */
+  const { pathname: previousPath } = useLocation()
+  const history = useHistory()
+
+  const preferenceOption = useRef('')
+
+  const handleOpenPreferences = useCallback(
+    screen => {
+      preferenceOption.current = screen
```

Comment:
I might rename all instances of `preferenceOption` to `preferenceScreen`, to align the argument names. "Option" is a bit overloaded when talking about preferences.